### ISSUE TITLE
User can send to channel they are not in via API.

### ISSUE THREAD

**Issue by setpill** [mattermost/mattermost-server#10525]
#### Summary
User can send to channel they are not in via API.

#### Steps to reproduce

mattermost version: 5.8.0
matterircd version: 0.19.1

1. Be in mattermost channel.
2. Connect to matterircd.
3. In mattermost UI, leave channel

#### Expected behavior

* IRC client no longer receives messages from channel.
* Sending to channel from IRC is not possible.
* IRC client leaves channel.

#### Observed behavior

* :heavy_check_mark: IRC client no longer receives messages from channel.
* :x: Sending to channel from IRC is *still* possible.
* :x: IRC client *does not* leave channel (matterircd issue).

#### Possible fixes

Check whether a user is in a channel before allowing to send messages to it.

**Comment by setpill** [comment]
Related issue: https://github.com/42wim/matterircd/issues/233

**Comment by setpill** [comment]
NB: this also goes for secret channels and channels and also works for getting kicked (rather than leaving voluntarily).

**Comment by amyblais** [comment]
Hi @setpill,

So far our developers have discussed that this behaviour is based on permissions. We still enforce permission checks, but we're not sure if we should allow a post from a user who is not in a channel. I'll see how the discussion continues.

**Comment by hanzei** [comment]
@amyblais is the discussion public and you can link to it?

**Comment by amyblais** [comment]
Opened a bug ticket here: https://mattermost.atlassian.net/browse/MM-14941.

**Comment by grundleborg** [comment]
@setpill were you using a System Admin account? In the default permissions configurations, System Admins can post to any channel on the entire system, regardless of membership. I haven't been able to replicate any issue with Mattermost allowing posting to channels when you don't have permission to.

**Comment by setpill** [comment]
@grundleborg I verified that A) I did have System Admin level access to our testing instance and B) I was not able to replicate this after being demoted. This solves the primary concern, but I will leave the issue open for further discussion about how this should function for System Admins. It is strange to me that this is possible through the API but not the webinterface, but I am not sure if this would enable desired usecases.
